### PR TITLE
New guide for connecting existing CDK backend resources to apps

### PR DESCRIPTION
**Pull request by hibler13**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Comment by hibler13** [comment]
@offlineprogrammer Here's the PR for the guide for your review.

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 1]
```diff
@@ -0,0 +1,865 @@
...
+mkdir cdk-backend 
+
+```
+
+**Step 2:** Navigate to the `cdk-backend` folder and create a new CDK project by running the `cdk init` command and specifying your preferred language.
+
+```bash showLineNumbers={false}
+
+cd cdk-backend 
+cdk init  --language typescript
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:66]:
```suggestion
cdk init --language typescript 
```

**Review by heatheramz — COMMENTED** [review]
@hibler13 For both the CDK and CLI guide I don't think we should be showing to android and swift users. Not sure on react native

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 2]
```diff
@@ -0,0 +1,865 @@
...
+* Flutter and its command-line tools [installed](https://docs.flutter.dev/get-started/install) and configured (if completing the second part of this guide).
+
+## Build a GraphQL API using the Amplify Data CDK construct
+
+The CDK provides a simple way to define cloud infrastructure in code. In this section, we will use the CDK to build out the backend resources for our application.
+
+**Step 1:** Create a folder for the CDK app by running the following command in your terminal. 
+
+```bash showLineNumbers={false}
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:56]: these newlines don't appear to affect the the display of the code snippet on the site, but for consistency with other pages can we remove the surrounding newlines in code snippets?

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 3]
```diff
@@ -0,0 +1,865 @@
...
+npm install @aws-amplify/data-construct
+
+```
+
+**Step 5:** Update the `cdk_backend/lib/cdk-backend-stack.ts` file as shown in the following code to use the `AmplifyData` construct  to create an AWS AppSync API. 
+
+```bash title="lib/cdk-backend-stack.ts"
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:82]:
```suggestion
```ts title="lib/cdk-backend-stack.ts"
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 4]
```diff
@@ -0,0 +1,865 @@
...
+* A text editor. For this guide, we will use VS Code, but you can use your preferred IDE.
+* Flutter and its command-line tools [installed](https://docs.flutter.dev/get-started/install) and configured (if completing the second part of this guide).
+
+## Build a GraphQL API using the Amplify Data CDK construct
+
+The CDK provides a simple way to define cloud infrastructure in code. In this section, we will use the CDK to build out the backend resources for our application.
+
+**Step 1:** Create a folder for the CDK app by running the following command in your terminal. 
+
+```bash showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:55]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

Comment (josefaidt) [reply src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:55]: add a "Terminal" title for additional context

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 5]
```diff
@@ -0,0 +1,865 @@
...
+
+mkdir cdk-backend 
+
+```
+
+**Step 2:** Navigate to the `cdk-backend` folder and create a new CDK project by running the `cdk init` command and specifying your preferred language.
+
+```bash showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:63]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 6]
```diff
@@ -0,0 +1,865 @@
...
+cd cdk-backend 
+cdk init  --language typescript 
+
+```
+
+**Step 3:** Open the newly created CDK project using VS Code, or your preferred IDE.
+
+**Step 4:** In your terminal, navigate to the `cdk_backend` root folder, and install the AWS Amplify Data package by running the following command.
+
+```bash showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:74]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 7]
```diff
@@ -0,0 +1,865 @@
...
+        defaultAuthorizationMode: 'API_KEY',
+        apiKeyConfig: {
+          expires: cdk.Duration.days(30)
+        }
+      }
+    });
+ // highlight-end
+  }
+}
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:112]:
```suggestion
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {
  AmplifyData,
  AmplifyDataDefinition
} from '@aws-amplify/data-construct';

export class CdkBackendStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // highlight-start
    new AmplifyData(this, 'AmplifyCdkData', {
      definition: AmplifyDataDefinition.fromString(/* GraphQL */ `
        type Todo @model @auth(rules: [{ allow: public }]) {
          id: ID!
          name: String!
          description: String
          complete: Boolean
        }
      `),
      authorizationModes: {
        defaultAuthorizationMode: 'API_KEY',
        apiKeyConfig: {
          expires: cdk.Duration.days(30)
        }
      }
    });
    // highlight-end
  }
}
```

ran this snippet through Prettier

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 8]
```diff
@@ -0,0 +1,865 @@
...
+Now that you have built the backend API with the CDK, you can connect a frontend. Refer to section two to connect a Flutter app or section three to connect a React app, or complete both to connect frontends in each framework.
+
+## Build a Flutter app and connect to the GraphQL API
+
+In this section, we will connect a Flutter web app to the GraphQL API we created with the CDK. First, we will initialize a Flutter project, define models matching our schema, and use Amplify to integrate CRUD operations. Then we will add UI pages to manage to-do items with queries and mutations. Finally, we will configure Amplify with our backend details and run the app to demonstrate full functionality with our existing API.
+
+**Step 1:** Create a Flutter app by running the following command in your terminal.
+
+```dart showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:139]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 9]
```diff
@@ -0,0 +1,865 @@
...
+
+flutter create flutter_todo_app --platforms=web   
+
+```
+
+**Step 2:** Open the newly created Flutter app by running the following commands in your terminal. 
+
+```dart
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:147]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 10]
```diff
@@ -0,0 +1,865 @@
...
+
+cd flutter_todo_app 
+code . -r
+
+```
+
+**Step 3**: Update the `pubspec.yaml` file in the app’s root directory to add the required dependencies, as shown in the following code.
+
+{/* cSpell:disable */}
+```dart title="pubspec.yaml"
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:157]:
```suggestion
```yaml title="pubspec.yaml"
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 11]
```diff
@@ -0,0 +1,865 @@
...
+```dart
+
+cd flutter_todo_app 
+code . -r
+
+```
+
+**Step 3**: Update the `pubspec.yaml` file in the app’s root directory to add the required dependencies, as shown in the following code.
+
+{/* cSpell:disable */}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:156]: is cspell tripping on `pubspec`? should we add this to the dictionary? https://github.com/aws-amplify/docs/blob/main/cspell.json

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 12]
```diff
@@ -0,0 +1,865 @@
...
+```
+{/* cSpell:enable */}
+
+**Step 4:** Run the following command in your terminal to install the dependencies you added to the `pubspec.yaml` file.
+
+```dart showLineNumbers={false}
+
+  flutter pub get 
+
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:194]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
flutter pub get
```
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 13]
```diff
@@ -0,0 +1,865 @@
...
+
+**Step 5:** Install the @aws-amplify/cli packages by running the following command. 
+
+```dart showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:198]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 14]
```diff
@@ -0,0 +1,865 @@
...
+
+```
+
+**Step 6:** Create a new folder and name it `graphql`. Inside it, create the file `schema.graphql`. 
+
+<img src="/images/existing-resources/schema-graphql-cdk.png" alt="The schema.graphql file inside the graphql folder." width="450" />
+
+**Step 7:** Update the file `schema.graphql`, as shown in the following example, to define the Todo data model, similar to what you used for the CDK app. 
+
+```dart title="schema.graphql"
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:210]:
```suggestion
```graphql title="schema.graphql"
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 15]
```diff
@@ -0,0 +1,865 @@
...
+
+```
+
+**Step 8:** Run the following command to generate the GraphQL client helper models inside the `lib/models` folder.
+
+```dart
+
+npx @aws-amplify/cli codegen models   --model-schema ./graphql   --target flutter   --output-dir ./lib/models
+
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:227]:
```suggestion
```bash
npx @aws-amplify/cli codegen models --model-schema ./graphql --target flutter --output-dir ./lib/models
```
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 16]
```diff
@@ -0,0 +1,865 @@
...
+      },
+    );
+  }
+}
+
+```
+
+**Step 12:** Run the app in the Chrome browser using the following command.
+
+```dart showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:644]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 17]
```diff
@@ -0,0 +1,865 @@
...
+
+Congratulations on successfully creating a new Flutter app and linking it to a preexisting GraphQL API built with AWS CDK. The next optional section will guide you through doing the same for a React web app. Alternatively, you can skip ahead to the ["Clean up resources"](/[platform]/build-a-backend/existing-resources/cdk/#clean-up-resources) section.
+
+## Build a React app and connect to the GraphQL API
+
+In this section, we will connect a React web app to our existing GraphQL API. First, we will create a new React project and install the necessary Amplify packages. Next, we will use the Amplify CLI to generate GraphQL code matching our API structure. Then, we will add React components to perform queries and mutations to manage to-do items in our API. After that, we will configure the Amplify library with details of our backend API. Finally, we will run the application to demonstrate full CRUD functionality with our existing API. 
+
+**Step 1:** Create a React app by running the following command in your terminal.
+
+```react showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:662]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 18]
```diff
@@ -0,0 +1,865 @@
...
+
+npx create-react-app react-amplify-connect 
+
+```
+
+**Step 2:** Open the newly created React app using VS Code, or your preferred IDE.
+ 
+**Step 3:** Install the aws-amplify, @aws-amplify/ui-react, and @aws-amplify/cli packages by running the following commands.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:670]:
```suggestion
**Step 3:** Install the `aws-amplify`, `@aws-amplify/ui-react`, and `@aws-amplify/cli` packages by running the following commands. 
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 19]
```diff
@@ -0,0 +1,865 @@
...
+
+npx create-react-app react-amplify-connect 
+
+```
+
+**Step 2:** Open the newly created React app using VS Code, or your preferred IDE.
+ 
+**Step 3:** Install the aws-amplify, @aws-amplify/ui-react, and @aws-amplify/cli packages by running the following commands. 
+
+```react showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:672]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 20]
```diff
@@ -0,0 +1,865 @@
...
+
+**Step 4:** Use the awsAppsyncApiId and awsAppsyncRegion values of the CDK stack you created previously to generate the GraphQL client helper code by running the following command.
+
+<img src="/images/existing-resources/outputs-cdk.png" alt="awsAppsyncApiId and awsAppsyncRegion values highlighted within the outputs of the CDK stack." width="750" />
+
+```react
+
+npx @aws-amplify/cli codegen add --apiId <awsAppsyncApiId> --region <awsAppsyncRegion> 
+
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:686]:
```suggestion
```react
npx @aws-amplify/cli codegen add --apiId <aws-appsync-api-id> --region <aws-appsync-region> 
```
```

revised casing of placeholder text

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 21]
```diff
@@ -0,0 +1,865 @@
...
+
+**Step 5:** Accept the default values for the prompts. 
+
+```react
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:690]:
```suggestion
```console title="Terminal" showLineNumbers={false}
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 22]
```diff
@@ -0,0 +1,865 @@
...
+
+```
+
+The Amplify CLI will create the GraphQL client helper code inside the `src/graphql` folder.
+
+<img src="/images/existing-resources/client-helper-models2-cdk.png" alt="mutations.js, queries.js, and subscriptions.js within the graphql folder." width="400" />
+
+**Step 6:** Update the `App.js` file with the following code to create a form with a button to create to-dos, as well as a way to fetch and render the to-do list.
+
+```react title="src/App.js"
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:710]:
```suggestion
```jsx title="src/App.js"
```

File: src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx [hunk 23]
```diff
@@ -0,0 +1,865 @@
...
+};
+
+export default App;
+
+```
+
+
+**Step 7:** Run the app using the following command.
+
+```react showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/existing-resources/cdk/index.mdx:839]:
```suggestion
```bash title="Terminal" showLineNumbers={false}
```

**Review by josefaidt — APPROVED** [review]

**Review by heatheramz — APPROVED** [review]
LGTM

**Pull request merged**